.buildkite/e2e/pipeline-gen/main.go

```diff
@@ -0,0 +1,390 @@
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package main
+
+import (
+	_ "embed"
+	"errors"
+	"flag"
+	"fmt"
+	"html/template"
+	"io"
+	"path/filepath"
+	"runtime"
+	"sort"
+
+	"math/rand"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"golang.org/x/exp/slices"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	EnvVarProvider             = "E2E_PROVIDER"
+	EnvVarK8sVersion           = "DEPLOYER_K8S_VERSION"
+	EnvVarStackVersion         = "E2E_STACK_VERSION"
+	EnvVarBuildkiteBuildNumber = "BUILDKITE_BUILD_NUMBER"
+	EnvVarBuildNumber          = "BUILD_NUMBER"
+	EnvVarPipeline             = "PIPELINE"
+	EnvVarClusterName          = "CLUSTER_NAME"
+	EnvVarTestOpts             = "TEST_OPTS"
+	EnvVarTestsMatch           = "TESTS_MATCH"
+	EnvVarBuildLicensePubkey   = "BUILD_LICENSE_PUBKEY"
+	EnvVarLicensePubKey        = "export LICENSE_PUBKEY"
+	EnvVarTestLicense          = "TEST_LICENSE"
+	EnvVarMonitoringSecrets    = "MONITORING_SECRETS"
+	EnvVarE2EJson              = "E2E_JSON"
+	EnvVarGoTags               = "GO_TAGS"
+	EnvVarOperatorImage        = "OPERATOR_IMAGE"
+	EnvVarE2EImage             = "E2E_IMG"
+)
+
+var (
+	//go:embed pipeline.tpl.yaml
+	pipelineTemplate string
+
+	// providersInDocker are k8s providers that require the deployer to run in Docker
+	providersInDocker = []string{"kind", "aks", "ocp", "tanzu"}
+	// providersNoCleanup are k8s providers that do not require the cluster to be deleted after use
+	providersNoCleanup = []string{"kind"}
+
+	semverRE = regexp.MustCompile(`\d*\.\d*\.\d*(-\w*)?`)
+	chars    = []rune("0123456789abcdefghijklmnopqrstuvwxyz")
+
+	shortcuts = map[string]string{
+		"p": EnvVarProvider,
+		"k": EnvVarK8sVersion,
+		"s": EnvVarStackVersion,
+		"t": EnvVarTestsMatch,
+	}
+
+	fixed string
+	mixed string
+
+	output string
+
+	rootDir string
+)
+
+func init() {
+	flag.StringVar(&fixed, "f", "", "Fixed variables (comma-separated list)")
+	flag.StringVar(&mixed, "m", "", "Mixed variables (comma-separated list of colon-separated list)")
+	flag.StringVar(&output, "o", "buildkite-pipeline", "Type of output: buildkite-pipeline or envfile")
+	flag.Parse()
+
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		fmt.Printf("failed to get current path")
+		os.Exit(1)
+	}
+	rootDir = filepath.Join(filepath.Dir(filename), "../../..")
+}
+
+func main() {
+	stat, err := os.Stdin.Stat()
+	handlErr("Failed to read stdin", err)
+
+	var groups []Group
+
+	// no stdin
+	if stat.Mode()&os.ModeCharDevice != 0 {
+
+		fixedEnv, err := stringListToEnv(fixed)
+		handlErr("Failed to read fixed variables", err)
+
+		mixedEnv, err := stringListToEnvs(mixed)
+		handlErr("Failed to read mixed variables", err)
+
+		groups = []Group{{
+			Fixed: fixedEnv,
+			Mixed: mixedEnv,
+		}}
+
+	} else {
+		in, err := io.ReadAll(os.Stdin)
```

Comment:
So when reading from a pipe all `-f` `-m` flags are ignored. Should we let the user know if case they specified some that they are without effect?